transports/sqs/delayed-delivery.md

```diff
@@ -0,0 +1,146 @@
+
+When a sender sends a delayed message to a destination it determines whether the delay is less or equal to 900 seconds. If that is the case the message is directly delayed to the destination queue by setting the `DelaySecond` behavior on the message. When the timeout is greater than 900 seconds the message is sent to the FIFO queue of the destination endpoint with a message attributed called `NServiceBus.AmazonSQS.DelaySeconds`. The delayed message consumer on the destination's FIFO queue receives all timeouts that are due after 900 seconds. When a timeout is due and the remaining delay is less or equal to 900 seconds the message is directly delayed to the destination queue. If the remaining delay is greater than 900 seconds the message is sent back to the FIFO queue containing a `DelaySeconds` attribute with the remaining timeout interval. The following sequence diagram illustrates the process:
+
+```mermaid
+sequenceDiagram
+    participant S as Sender
+    participant D as Destination
+    participant F as Destination-delay.fifo
+    alt delay <= 900sec
+        S ->> D: Message DelaySecond = delay
+    else delay > 900sec
+        S -->> F:  Attribute DelaySecond = delay
+        loop every 900sec
+           alt remaining delay <= 900sec
+              F ->> D: Message DelaySecond = remaining delay
+           else remaining delay> 900sec
+              F -->> F:  Attribute DelaySecond = remaining delay
+           end
+        end
+end
+```
+
+### Clock drift
+
+In order to avoid clock drift the broker timestamps are used whereever possible to calculate the remaining timeout. The due time calculation uses `SentTimestamp` as well as `ApproximateFirstReceiveTimestamp` set by the broker. Only in cases of redelivery when `ApproximateReceiveCount` is higher than one the clients clock is used and thus subjected to clock drift.
+
+### Delivery
+
+For unrestricted delayed deliveries the last step is always a handover from the FIFO queue to the endpoint's input queue. SQS does not provide cross queue operation transactions thus the handover is subjected to retries. In cases of retries it might be possible that timeouts are delivered more than once. Message handlers need to be idempotent when used with transports with [transaction](transports/transactions.md) level `Receive Only` or below. The following diagram illustrates that:
+
+```mermaid
+sequenceDiagram
+    participant D as Destination
+    participant F as Destination-delay.fifo
+    F ->>+ F: Timeout due
+    F ->> D: Send with remaing delay
+    Note left of D: Original
+    F ->>- F: Delete Delayed Message failed
+    Note right of F: i.ex. network outage
+    F ->>+ F: Timeout due
+    F ->> D: Send with remaing delay
+    F ->>- F: Delete Delayed Message
+    Note left of D: Duplicate
+```
+
+### Example
+
+Below an example with a delayed delivery less or equal of 15 min.
+
+```mermaid
+graph LR
```

Comment:
![image](https://user-images.githubusercontent.com/174258/37403346-cd9447ca-278e-11e8-9ddb-c7dfef4afbb4.png)

Reply:
Online editor doesn't render font-awesome. fa:fa-hourglass-half should be properly rendered on the docs site